edge-hub/core/src/Microsoft.Azure.Devices.Edge.Hub.Amqp/CbsNode.cs

```diff
@@ -73,24 +74,42 @@ public void RegisterLink(IAmqpLink link)
             Events.LinkRegistered(link);
         }
 
-        public async Task<bool> AuthenticateAsync(string id, Option<string> modelId)
+        public async Task<bool> AuthenticateAsync(string id, Option<string> modelId, Option<string> authChain)
```

Comment:
Probably fine for now, but should we consider passing in some sort object that contains modelId, authChain, and whatever else we may add in the future?

Reply:
Hmm....model ID and authchain aren't logically related in any way though.  I guess maybe if the param-list gets really long in the future?  For now there's only 3 so I think it's okay.

Reply:
We should also pass in the productInfo in here. 
Check with @ancaantochi on how to get that?